elements/pl-symbolic-input/pl-symbolic-input.py

```diff
@@ -367,63 +369,100 @@ def grade(element_html, data):
         data['partial_scores'][name] = {'score': 0, 'weight': weight}
 
 
-def test(element_html, data):
+def test(element_html: str, data: pl.ElementTestData) -> None:
     element = lxml.html.fragment_fromstring(element_html)
     name = pl.get_string_attrib(element, 'answers-name')
+    variables = get_variables_list(pl.get_string_attrib(element, 'variables', VARIABLES_DEFAULT))
+    allow_complex = pl.get_boolean_attrib(element, 'allow-complex', ALLOW_COMPLEX_DEFAULT)
     weight = pl.get_integer_attrib(element, 'weight', WEIGHT_DEFAULT)
+    imaginary_unit = pl.get_string_attrib(element, 'imaginary-unit-for-display', IMAGINARY_UNIT_FOR_DISPLAY_DEFAULT)
+
+    # Get raw correct answer
+    a_tru = data['correct_answers'][name]
+
+    # Parse correct answer based on type
+    if isinstance(a_tru, str):
+        a_tru = phs.convert_string_to_sympy(a_tru, variables, allow_complex=allow_complex)
+    else:
+        a_tru = phs.json_to_sympy(a_tru, allow_complex=allow_complex)
+
+    # Substitute in imaginary unit symbol
+    a_tru_str = str(a_tru.subs(sympy.I, sympy.Symbol(imaginary_unit)))
 
     result = data['test_type']
     if result == 'correct':
-        data['raw_submitted_answers'][name] = str(pl.from_json(data['correct_answers'][name]))
+        data['raw_submitted_answers'][name] = a_tru_str
         data['partial_scores'][name] = {'score': 1, 'weight': weight}
+
     elif result == 'incorrect':
-        data['raw_submitted_answers'][name] = str(pl.from_json(data['correct_answers'][name])) + ' + {:d}'.format(random.randint(1, 100))
+        data['raw_submitted_answers'][name] = f'{a_tru_str} + {random.randint(1, 100):d}'
         data['partial_scores'][name] = {'score': 0, 'weight': weight}
+
     elif result == 'invalid':
-        invalid_type = random.choice(['float', 'complex', 'expression', 'function', 'variable', 'syntax', 'escape', 'comment'])
+        invalid_type_choices: List[Literal['float', 'complex', 'expression', 'function', 'variable', 'syntax', 'escape', 'comment']] = \
+            ['float', 'complex', 'expression', 'function', 'variable', 'syntax', 'escape', 'comment']
+
+        invalid_type = random.choice(invalid_type_choices)
+
         if invalid_type == 'float':
-            data['raw_submitted_answers'][name] = 'x + 1.234'
-            s = 'Your answer contains the floating-point number ' + str(1.234) + '. '
-            s += 'All numbers must be expressed as integers (or ratios of integers). '
-            s += '<br><br><pre>' + phs.point_to_error('x + 1.234', 4) + '</pre>'
-            data['format_errors'][name] = s
+            invalid_input = 'x + 1.234'
+            data['raw_submitted_answers'][name] = invalid_input
+            data['format_errors'][name] = (
```

Comment:
@nwalters512 When I was going over this test function, I noticed that changing this format error message didn't cause any test failures. Are the format errors produced not checked against the reference ones here? If so, I can remove these from the tests (since they're not necessary).

Reply:
It looks like we only compare the keys, not their values:

https://github.com/PrairieLearn/PrairieLearn/blob/1957d08a566f18c29519562ed716e5c820b8988a/lib/question.js#L901-L905

@mwest1066 do you know why that is?

Reply:
That makes sense if we don't care about the error message (as long as their is one). It might be nice in this case to check the values as well, but could cause issues testing other elements.